base/bucket_gocb.go

```diff
@@ -978,6 +978,45 @@ func (bucket CouchbaseBucketGoCB) WriteCasWithXattr(k string, xattrKey string, e
 	return cas, err
 }
 
+// CAS-safe delete of a document, and update of it's corresponding xattr
+func (bucket CouchbaseBucketGoCB) DeleteAndUpdateXattr(k string, xattrKey string, exp int, cas uint64, xv interface{}) (casOut uint64, err error) {
+
+	// WriteCasWithXattr always stamps the xattr with the new cas using macro expansion, into a top-level property called 'cas'.
+	// This is the only use case for macro expansion today - if more cases turn up, should change the sg-bucket API to handle this more generically.
+	xattrCasProperty := fmt.Sprintf("%s.cas", xattrKey)
+	worker := func() (shouldRetry bool, err error, value interface{}) {
+
+		docFragment, removeErr := bucket.Bucket.MutateInEx(k, gocb.SubdocDocFlagNone, gocb.Cas(cas), uint32(0)).
```

Comment:
We should be setting the expiry value in the MutateInEx call - instead of uint32(0) it should be set to the incoming expiry value (i.e. uint32(exp)).